pkg/render/node.go

```diff
@@ -1433,6 +1474,52 @@ func (c *nodeComponent) nodePodSecurityPolicy() *policyv1beta1.PodSecurityPolicy
 	return psp
 }
 
+// hostPathInitContainer creates an init container that changes the permissions on hostPath volumes
+// so that they can be written to by a non-root container.
+func (c *nodeComponent) hostPathInitContainer() corev1.Container {
+	rootUID := int64(0)
+	mounts := []corev1.VolumeMount{
+		{
+			MountPath: "/var/run",
+			Name:      "var-run",
+			ReadOnly:  false,
+		},
+		{
+			MountPath: "/var/lib",
+			Name:      "var-lib",
+			ReadOnly:  false,
+		},
+	}
+
+	return corev1.Container{
+		Name:  "hostpath-init",
+		Image: c.nodeImage,
+		Env: []corev1.EnvVar{
+			{Name: "NODE_USER_ID", Value: "1000"},
+		},
+		VolumeMounts: mounts,
+		SecurityContext: &corev1.SecurityContext{
+			RunAsUser: &rootUID,
+		},
+		Command: []string{"sh", "-c", "calico-node -hostpath-init"},
+	}
+}
+
+// runAsNonPrivileged checks to ensure that all of the proper installation values are set for running
+// Calico as non privileged.
+func (c *nodeComponent) runAsNonPrivileged() bool {
+	// Check that the NonPrivileged flag is set
+	nonPrivileged := c.cfg.Installation.NonPrivileged != nil && *c.cfg.Installation.NonPrivileged == operatorv1.NonPrivilegedEnabled
+
+	// Check that BGP is not enabled
+	noBGP := !bgpEnabled(c.cfg.Installation)
+
+	// Only allowed to run as non privileged for OS Calico
+	variantCalico := c.cfg.Installation.Variant != operatorv1.TigeraSecureEnterprise
```

Comment:
This check should not be needed here. There should be validation done by the core_controller, the validation code should be added to https://github.com/tigera/operator/blob/master/pkg/controller/installation/validation.go#L30.
The result of specifying NonPrivileged and Enterprise should be a degraded status on the calico tigerastatus (which is where a validation failure that I linked to would show up).